hubspot/connection/__init__.py

```diff
@@ -228,11 +230,11 @@ def __exit__(self, exc_type, exc_value, traceback):
         self._session.close()
 
 
-_AuthenticationKey = Record.create_type('_AuthenticationKey', 'key_value')
+_AuthenticationKey = Record.create_type(str('_AuthenticationKey'), str('key_value'))
```

Comment:
Without the str(), and with the additions that make it python 3 compatibility, you end up with this kind of error:

Traceback (most recent call last):
  File "/home/jtprince/virtualenvs/python2.7-playground/local/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/jtprince/virtualenvs/python2.7-playground/local/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/jtprince/virtualenvs/python2.7-playground/local/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/jtprince/dev/hubspot-connection/tests/test_testing_utils.py", line 24, in <module>
    from hubspot.connection.exc import HubspotAuthenticationError
  File "/home/jtprince/dev/hubspot-connection/hubspot/connection/__init__.py", line 233, in <module>
    _AuthenticationKey = Record.create_type('_AuthenticationKey', 'key_value')
  File "build/bdist.linux-x86_64/egg/pyrecord/__init__.py", line 204, in create_type
    **default_values_by_field_name
  File "build/bdist.linux-x86_64/egg/pyrecord/__init__.py", line 239, in extend_type
    default_values_by_field_name,
  File "build/bdist.linux-x86_64/egg/pyrecord/__init__.py", line 245, in _create_type
    record_type = type(type_name, (cls,), {})
TypeError: type() argument 1 must be string, not unicode

str() solves the issue by giving pyrecord a bytestring instead of unicode under python 2.7 and giving  pyrecord unicode instead of a bytestring in python 3.4.  I believe it is the right way to solve this problem while achieving compatibility (although I'm always open to suggestions, especially when it's your code base ;)